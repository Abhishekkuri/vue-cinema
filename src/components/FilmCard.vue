<template>
  <div class="card-container">
    <div class="card">
      <div class="poster">
        <img :src="film.Poster" alt="Poster Not Available" />
      </div>
      <div class="card-detail">
        <h4 class="card-detail-item">{{ film.Title }}</h4>
        <div class="card-detail-item">Type: {{ film.Type }}</div>
        <div class="card-detail-item">Rated: {{ film.Rated }}</div>
        <div card-detail-item>
          Genre:
          <span
            class="custom-span"
            v-for="genre in film.Genre.split(',')"
            :key="genre"
          >
            {{ genre }}
          </span>
        </div>
        <div class="card-detail-item">Year: {{ film.Year }}</div>
        <div class="card-detail-item">
          Actors:
          <span
            class="custom-span actor"
            v-for="actor in film.Actors.split(',')"
            :key="actor"
          >
            {{ actor }}
          </span>
        </div>
        <div class="card-detail-item">Directors: {{ film.Director }}</div>
        <div class="card-detail-item">Awards: {{ film.Awards }}</div>
        <p class="card-detail-item">
          Plot:
          <span>{{ filmPlot }}</span>
          <a
            class="read-more"
            v-if="readMoreActivated && film.Plot.length > 200"
            @click.prevent="toggleReadMore"
          >
            read more...
          </a>
          <a
            class="read-more"
            v-if="!readMoreActivated && film.Plot.length > 200"
            @click.prevent="toggleReadMore"
          >
            read less
          </a>
        </p>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: ["film"],
  data() {
    return {
      readMoreActivated: this.film.Plot.length > 200,
    };
  },
  computed: {
    filmPlot() {
      return this.readMoreActivated
        ? this.film.Plot.slice(0, 200)
        : this.film.Plot;
    },
  },
  methods: {
    toggleReadMore() {
      this.readMoreActivated = !this.readMoreActivated;
    },
  },
};
</script>

<style lang="scss" scoped>
.card-container {
  padding: 1rem;
  margin: 1rem;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  background-color: $color-secondary;
  &:hover {
    box-shadow: 0 8px 16px 0 $color-primary;
  }
  .poster {
    padding: 5px;
    height: 28rem;
    img {
      width: 100%;
      height: 28rem;
      border-radius: $color-grey-dark;
    }
  }
  .card-detail {
    padding: 5px;
    .card-detail-item {
      padding: 3px 0;
      .read-more {
        text-decoration: none;
        cursor: pointer;
        color: $color-secondary;
      }
    }
  }
  .custom-span {
    border: 1px solid $color-grey-dark;
    display: inline-block;
    margin-right: 5px;
    margin-bottom: 5px;
    padding: 0 3px;
    border-radius: 5px;
  }
}

@media screen and (max-width: 820px) {
}
</style>
