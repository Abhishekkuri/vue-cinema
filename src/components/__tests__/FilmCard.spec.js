import { shallowMount } from "@vue/test-utils";
import FilmCard from "@/components/FilmCard";
import merge from "lodash.merge";

const defaultMountingOptions = {
  propsData: {
    film: {
      Title: "Some Title",
      Type: "Series",
      Rated: "8",
      Actors: "Ben, John, Henk",
      Genre: "Mystry, Drama, Musical",
      Plot: "This is a movie plot with more than 200 words. This is a movie plot with more than 200 words. This is a movie plot with more than 200 words. This is a movie plot with more than 200 words. This is a movie plot with more than 200 words. This is a movie plot with more than 200 words.",
      Director: "Ben, John, Henk",
      Awards: "Oscar, BAFTA",
    },
  },
};

function createWrapper(overrides) {
  return shallowMount(FilmCard, merge(defaultMountingOptions, overrides));
}

describe("FilmCard.vue", () => {
  test("FilmCard renders correctly", () => {
    const wrapper = createWrapper();
    expect(wrapper.element).toMatchSnapshot();
  });
  test("renders the Poster", () => {
    const wrapper = createWrapper();
    expect(wrapper.find("img").exists()).toBe(true);
  });
  test("renders the Title", () => {
    const wrapper = createWrapper();
    expect(wrapper.text()).toContain(
      defaultMountingOptions.propsData.film.Title
    );
  });
  test("renders the Type", () => {
    const wrapper = createWrapper();
    expect(wrapper.text()).toContain(
      defaultMountingOptions.propsData.film.Type
    );
  });
  test("renders the Genre", () => {
    const wrapper = createWrapper();
    defaultMountingOptions.propsData.film.Genre.split(",").forEach((genre) => {
      expect(wrapper.text()).toContain(genre);
    });
  });
  test("renders the Actors", () => {
    const wrapper = createWrapper();
    defaultMountingOptions.propsData.film.Actors.split(",").forEach((actor) => {
      expect(wrapper.text()).toContain(actor);
    });
  });
  test("renders the Directors", () => {
    const wrapper = createWrapper();
    expect(wrapper.text()).toContain(
      defaultMountingOptions.propsData.film.Director
    );
  });
  test("renders the Awards", () => {
    const wrapper = createWrapper();
    expect(wrapper.text()).toContain(
      defaultMountingOptions.propsData.film.Awards
    );
  });
  test("renders the Plot", () => {
    const wrapper = createWrapper();
    expect(wrapper.text()).toContain(
      defaultMountingOptions.propsData.film.Plot.slice(0, 200)
    );
  });
  test("renders the Read More", () => {
    const wrapper = createWrapper();
    expect(wrapper.contains(".read-more")).toBe(true);
  });
});
