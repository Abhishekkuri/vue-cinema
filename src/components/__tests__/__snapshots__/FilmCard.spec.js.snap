// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard.vue FilmCard renders correctly 1`] = `
<div
  class="card-container"
>
  <div
    class="card"
  >
    <div
      class="poster"
    >
      <img
        alt="Poster Not Available"
      />
    </div>
     
    <div
      class="card-detail"
    >
      <h4
        class="card-detail-item"
      >
        Some Title
      </h4>
       
      <div
        class="card-detail-item"
      >
        Type: Series
      </div>
       
      <div
        class="card-detail-item"
      >
        Rated: 8
      </div>
       
      <div
        card-detail-item=""
      >
        
        Genre:
        
        <span
          class="custom-span"
        >
          
          Mystry
        
        </span>
        <span
          class="custom-span"
        >
          
           Drama
        
        </span>
        <span
          class="custom-span"
        >
          
           Musical
        
        </span>
      </div>
       
      <div
        class="card-detail-item"
      >
        Year: 
      </div>
       
      <div
        class="card-detail-item"
      >
        
        Actors:
        
        <span
          class="custom-span actor"
        >
          
          Ben
        
        </span>
        <span
          class="custom-span actor"
        >
          
           John
        
        </span>
        <span
          class="custom-span actor"
        >
          
           Henk
        
        </span>
      </div>
       
      <div
        class="card-detail-item"
      >
        Directors: Ben, John, Henk
      </div>
       
      <div
        class="card-detail-item"
      >
        Awards: Oscar, BAFTA
      </div>
       
      <p
        class="card-detail-item"
      >
        
        Plot:
        
        <span>
          This is a movie plot with more than 200 words. This is a movie plot with more than 200 words. This is a movie plot with more than 200 words. This is a movie plot with more than 200 words. This is a mo
        </span>
         
        <a
          class="read-more"
        >
          
          read more...
        
        </a>
         
        <!---->
      </p>
    </div>
  </div>
</div>
`;
